### PR TITLE
Osman Elsahib-Database-Week-1

### PR DESCRIPTION
**Pull request by elsahib**
Hello Mentors,

Please review my HW submission.

### CODE REVIEW

File: week-1/mandatory/2-classes-db/task.md [hunk 1]
```diff
@@ -2,13 +2,21 @@
...
 To submit this homework write the correct commands for each question here:
 
 ```sql
-
-
+1. select * from rooms where rate > 100;
+2. select * from reservations where (checkin_date >= '2020-09-01' and checkin_date <='2020-09-30') and checkout_date - checkin_date < 3;
+3. select * from customers where city like 'M%';
+4. insert into room_types values('PENTHOUSE','185.00');
+5. insert into rooms (room_no, rate, room_type) values(501,(select def_rate from room_types where room_type = 'PENTHOUSE'),'PENTHOUSE'),(502,(select def_rate from room_types where room_type = 'PENTHOUSE'),'PENTHOUSE');
```
Comment (KeithBremer) [inline week-1/mandatory/2-classes-db/task.md:14]: Woo - very good use of subqueries, but sadly, we don't have time to cover those in this course.  Well done!

File: week-1/mandatory/2-classes-db/task.md [hunk 2]
```diff
@@ -2,13 +2,21 @@
...
+6. insert into rooms (room_no, rate, room_type) values(503,143.00,'PREMIER PLUS');
+7. select all from reservations where checkin_date >= '2020-08-01' and checkin_date <= '2020-08-31'; / select count(id) from reservations where checkin_date >= '2020-08-01' and checkin_date <= '2020-08-31';
```
Comment (KeithBremer) [inline week-1/mandatory/2-classes-db/task.md:16]: Not sure what you expect with `select all from ...` - it doesn't produce any useful results.
To catch reservations that overlap the ends of the month you could try using:
`... WHERE checkin_date <= '2020-08-31` and checkout_date > '2020-08-01'

File: week-1/mandatory/2-classes-db/task.md [hunk 3]
```diff
@@ -2,13 +2,21 @@
...
+8. select sum(checkout_date - checkin_date) from reservations where room_no between 200 and 299;
+9. select sum(total),avg(total) from invoices where total > 300;
+10. select sum(case when room_no between 100 and 199 then checkout_date - checkin_date end) as first_floor, sum(case when room_no between 200 and 299 then checkout_date - checkin_date end) as second_floor, sum(case when room_no between 300 and 399 then checkout_date - checkin_date end) as third_floor from reservations;
```
Comment (KeithBremer) [inline week-1/mandatory/2-classes-db/task.md]: Wow!  Very innovative and interesting use of the CASE construct!  Very good - you've clearly done more reading than was asked for - well done.

File: week-1/mandatory/2-classes-db/task.md (hunk 2 above)
Comment (elsahib) [reply week-1/mandatory/2-classes-db/task.md:16]: Oh, yes that's more like it. I never thought of using the checkout date as well in my where clause.
Thanks for that.

File: week-1/mandatory/2-classes-db/task.md (hunk 1 above)
Comment (elsahib) [reply week-1/mandatory/2-classes-db/task.md:14]: Thak you very much

File: week-1/mandatory/2-classes-db/task.md (hunk 3 above)
Comment (elsahib) [reply week-1/mandatory/2-classes-db/task.md]: Thanks a lot

**Comment by elsahib** [comment]
Thanks for everything Keith,

**Pull request closed without merging**